tests/unit/commands/local/lib/test_stack_provider.py

```diff
@@ -168,3 +168,43 @@ def test_remote_stack_is_skipped(self, resource_type, location_property_name):
                 Stack("", "", self.template_file, None, template),
             ],
         )
+
+    @parameterized.expand(
+        [
+            ("/path/to/template.yaml", AWS_SERVERLESS_APPLICATION, "Location", "./child.yaml", "/path/to/child.yaml"),
+            ("/path/to/template.yaml", AWS_SERVERLESS_APPLICATION, "Location", "child.yaml", "/path/to/child.yaml"),
+            ("somedir/template.yaml", AWS_CLOUDFORMATION_STACK, "TemplateURL", "./child.yaml", "somedir/child.yaml"),
+            ("somedir/template.yaml", AWS_CLOUDFORMATION_STACK, "TemplateURL", "child.yaml", "somedir/child.yaml"),
+            ("/path/to/template.yaml", AWS_SERVERLESS_APPLICATION, "Location", "file:///child.yaml", "/child.yaml"),
+            ("somedir/template.yaml", AWS_CLOUDFORMATION_STACK, "TemplateURL", "file:///child.yaml", "/child.yaml"),
+        ]
+    )
+    def test_sam_nested_stack_template_path_can_be_resolved_if_root_template_is_not_in_working_dir(
+        self, template_file, resource_type, location_property_name, child_location, child_location_path
+    ):
+        template = {
+            "Resources": {
+                "ChildStack": {
+                    "Type": resource_type,
+                    "Properties": {location_property_name: child_location},
+                }
+            }
+        }
+        self.get_template_data_mock.side_effect = lambda t: {
+            template_file: template,
+            child_location_path: LEAF_TEMPLATE,
+        }.get(t)
```

Comment:
What kind of sorcery happening here? :)

Reply:
I found this is the most simple way to "return Y if input is X" 😄